### PR TITLE
Detect directory entry modifications to uid/gid

### LINKED ISSUE #3435: DirectoryEntryBase::CompareTo() does not perform full comparison
https://github.com/cvmfs/cvmfs/blob/6bda271743101007b9735b80324db848f5beecb9/cvmfs/directory_entry.cc#L9-L40

This function will miss isolated changes to uid, gid, xattrs, chunks. This isn't an issue usually, as such changes when made via an overlayfs transaction will also change the mtime. It causes a problem if the changes are being made programmatically.

It also results in changed behaviour when publishing via a gateway vs not.

### PR DESCRIPTION
**Pull request by jblomer**
Partially fixes #3435: uid/gid changes are now properly handled

### CODE REVIEW

**Comment by jblomer** [comment]
@cernvm-bot unittest

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Comment by vvolkl** [comment]
@cernvm-bot  cloudtest

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

File: cvmfs/swissknife_check.cc [hunk 1]
```diff
@@ -128,6 +128,16 @@ bool CommandCheck::CompareEntries(const catalog::DirectoryEntry &a,
...
+  if (diffs & Difference::kUid) {
+    LogCvmfs(kLogCvmfs, kLogStderr, "uids differ: %d / %d (%s / %s)",
+             a.uid(), b.uid(), a.name().c_str(), b.name().c_str());
+    retval = false;
+  }
+  if (diffs & Difference::kGid) {
+    LogCvmfs(kLogCvmfs, kLogStderr, "gids differ: %d / %d (%s / %s)",
+             a.gid(), b.gid(), a.name().c_str(), b.name().c_str());
+    retval = false;
+  }
```
Comment (vvolkl) [inline cvmfs/swissknife_check.cc]:
```suggestion
  if (!is_transition_point) {
      if (diffs & Difference::kUid) {
        LogCvmfs(kLogCvmfs, kLogStderr, "uids differ: %d / %d (%s / %s)",
                 a.uid(), b.uid(), a.name().c_str(), b.name().c_str());
        retval = false;
      }
      if (diffs & Difference::kGid) {
        LogCvmfs(kLogCvmfs, kLogStderr, "gids differ: %d / %d (%s / %s)",
                 a.gid(), b.gid(), a.name().c_str(), b.name().c_str());
        retval = false;
      }
    }
```
This check should not should fail for nested catalog transition points. There could be a `catalog-chown` that is not expected to affect them, I think.

**Review by vvolkl — CHANGES_REQUESTED** [review]

File: cvmfs/swissknife_check.cc (hunk 1 above)
Comment (vvolkl) [reply cvmfs/swissknife_check.cc]: Commiting in order to start cloudtests

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Review by vvolkl — APPROVED** [review]

**Pull request merged**